src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OtlpCommonExtensions.cs

```diff
@@ -0,0 +1,116 @@
+
+            switch (kvp.Value)
+            {
+                case char:
+                case string:
+                    attrib.Value.StringValue = Convert.ToString(kvp.Value);
+                    break;
+                case bool b:
+                    attrib.Value.BoolValue = b;
+                    break;
+                case byte:
+                case sbyte:
+                case short:
+                case ushort:
+                case int:
+                case uint:
+                case long:
+                    attrib.Value.IntValue = Convert.ToInt64(kvp.Value);
+                    break;
+                case float:
+                case double:
+                    attrib.Value.DoubleValue = Convert.ToDouble(kvp.Value);
+                    break;
+                case Array array:
+                    var arrayValue = attrib.Value.ArrayValue = new OtlpCommon.ArrayValue();
+                    switch (kvp.Value)
+                    {
+                        case char[]:
+                        case string[]:
+                            foreach (var item in array)
+                            {
+                                arrayValue.Values.Add(new OtlpCommon.AnyValue { StringValue = Convert.ToString(item) });
+                            }
+
+                            break;
+                        case bool[]:
+                            foreach (var item in array)
+                            {
+                                arrayValue.Values.Add(new OtlpCommon.AnyValue { BoolValue = Convert.ToBoolean(item) });
+                            }
+
+                            break;
+                        case byte[]:
+                        case sbyte[]:
+                        case short[]:
+                        case ushort[]:
+                        case int[]:
+                        case uint[]:
+                        case long[]:
+                            foreach (var item in array)
+                            {
+                                arrayValue.Values.Add(new OtlpCommon.AnyValue { IntValue = Convert.ToInt64(item) });
+                            }
+
+                            break;
+                        case float[]:
+                        case double[]:
+                            foreach (var item in array)
+                            {
+                                arrayValue.Values.Add(new OtlpCommon.AnyValue { DoubleValue = Convert.ToDouble(item) });
+                            }
+
+                            break;
+                        default:
+                            return null;
+                    }
+
+                    break;
+
+                // case nint:    Pointer type.
+                // case nuint:   Pointer type.
+                // case ulong:   May throw an exception on overflow.
+                // case decimal: Converting to double produces rounding errors.
+                default:
+                    return null;
```

Comment:
this would be a breaking change? as we were doing ToString() before, and now we ignore the value? should we retain the current behavior?
(Or we treat current behavior as a unintentional bug, and this is indeed the fix)

Either way - please add to changelog entry as well.

Reply:
> Or we treat current behavior as a unintentional bug, and this is indeed the fix

This is where I'm leaning but want to hear other opinions. While unusual, I have seen a `ToString` overload be susceptible to exceptions, mutating state, and even deadlock.

As an end user, I'd prefer to have to be intentional about how I'm ToStringing my attributes that are not just simple numeric types or strings.

Reply:
I think this is the right change.

One thing to note is - do we log anywhere about this? i.e if I miss some values, and i enable self-diagnostics, would there be something that tells me exporter is dropping things?

Reply:
At one point, I thought the Jaeger and Zipkin exporters did not ToString arbitrary attributes. I guess they do now. So, I'm ok `ToString`ing everything else if that's what folks prefer.

Reply:
> One thing to note is - do we log anywhere about this? i.e if I miss some values, and i enable self-diagnostics, would there be something that tells me exporter is dropping things?

No logging at the moment - will add.

Reply:
> I think this is the right change.

In that case, we should update Zipkin, Jaeger, Console, Prometheus to behave the same. Maybe there's some way to unify this logic, but I haven't put my mind to this yet.

Reply:
From SIG meet: revert to use ToString(), and explore IFormattable in a follow up.